dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/common/DefaultIdentifiableObjectManager.java

```diff
@@ -1239,66 +1254,40 @@ public void removeUserGroupFromSharing( String userGroupUid )
     }
 
     @SuppressWarnings( "unchecked" )
-    private <T extends IdentifiableObject> IdentifiableObjectStore<T> getIdentifiableObjectStore(
-        Class<T> type )
+    private <T extends IdentifiableObject> IdentifiableObjectStore<T> getIdentifiableObjectStore( @Nonnull T object )
     {
-        initMaps();
-
-        IdentifiableObjectStore<? extends IdentifiableObject> store = identifiableObjectStoreMap.get( type );
-
-        if ( store == null )
-        {
-            store = identifiableObjectStoreMap.get( type.getSuperclass() );
```

Comment:
Note the odd `type.getSuperclass()`. Presumably this was to "unproxy" the `Class` but this is already done by the callers of the `getIdentifiableObjectStore(Class)` method. There is a `getIdentifiableObjectStore(Object)` method which uses `getRealClass(obj)`. So the cleaned code uses `type` not  `type.getSuperclass()` as lookup to find the corresponding store.